testsuite/tests/basic/patmatch.ml

```diff
@@ -149,8 +149,8 @@ let () =
 
 (* No string sharing PR#6322 *)
 let test x = match x with
-  | true -> "a"
-  | false -> "a"
+  | true -> Bytes.of_string "a"
+  | false -> Bytes.of_string "a"
```

Comment:
I think it does, but the test is rather pointless in `-safe-string` mode anyway.